lib/service/service.go

```diff
@@ -1588,12 +1588,14 @@ func (process *TeleportProcess) initSSH() error {
 				warnOnErr(s.Shutdown(payloadContext(payload)))
 			}
 		}
-		if conn.UseTunnel() {
+		if conn != nil && conn.UseTunnel() {
 			agentPool.Stop()
 		}
 
-		// Close BPF service.
-		warnOnErr(ebpf.Close())
+		if ebpf != nil {
+			// Close BPF service.
+			warnOnErr(ebpf.Close())
+		}
```

Comment:
Are the changes to this file related to this PR? It looks like it might be mixed in from another change set.

Reply:
It is part of the PR.
When you spin up a node and it is stuck retrying to join the cluster, killing it (ctrl-c) causes a nil pointer panic. I noticed it while reproducing the issue and rolled up the fix.

Happy to split it into a separate PR if you think that's best.

Reply:
No, it’s fine, just seemed unrelated!